core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@kzantow it just a marker, probably somebody else will suggest something better.